pitop/robotics/navigation/navigation_controller.py

```diff
@@ -129,6 +128,10 @@ def go_to(
 
         return self
 
+    def __exit__(self, exc_type, exc_value, exc_traceback):
+        if self._nav_thread.is_alive():
```

Comment:
should you set `self.navigator._stop_triggered = True` or something here to ensure this thread will actually join and allow the process to exit?